File: backend/app/models/ingredient.py
from datetime import UTC, datetime
from typing import Literal, Optional
from uuid import uuid4

from pydantic import BaseModel, Field


class Ingredient(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid4()))
    name: str
    quantity: str
    unit: Optional[str] = None
    raw_input: str
    confidence: float = Field(ge=0.0, le=1.0)
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))


class IngredientSession(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid4()))
    user_id: str
    ingredients: list[Ingredient] = Field(default_factory=list)
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    status: Literal["in_progress", "confirmed", "used"] = "in_progress"

File: frontend/src/types/ingredient.ts
/**
 * TypeScript types for ingredient-related data structures.
 * These types match the backend Pydantic models in backend/app/models/ingredient.py
 */

export interface Ingredient {
  id: string;
  name: string;
  quantity: string;
  unit?: string;
  raw_input: string;
  confidence: number;
  created_at: string;
}

export interface IngredientSession {
  id: string;
  user_id: string;
  ingredients: Ingredient[];
  created_at: string;
  updated_at: string;
  status: 'in_progress' | 'confirmed' | 'used';
}

export interface ParseResponse {
  ingredients: Ingredient[];
}
